encore.cabal

```diff
@@ -30,16 +30,16 @@ executable encorec
                   , CPP
                   , ScopedTypeVariables
   build-depends: MissingH
-               , base >=4.7 && <4.9
+               , base
                , containers >= 0.5.5.1
-               , directory >=1.2 && <1.3
+               , directory
                , hashable
                , hpc
                , mtl ==2.2.*
                , megaparsec >= 5.1.2
```

Comment:
I would suggest setting this dependency to megaparsec < 5.2
Since 5.3.1 is included as default by the lts, and 5.2 and above breaks compatibility.

That will also make stack solver not recommend deleting the extra-dep in stack.yaml
(Note, if using an unsupported cabal-install *stack solver* will error out due stack can't parse the output but it will compile fine. running with supported cabal-install-1.24 which is in lts-9.17 will successfully determine a build plan)